File: day-1/go/main.go
package main;

import (
  "bufio"
  "fmt"
  "log"
  "strconv"
  "os"
  "math"
)

func main() {
  masses := getInput()

  part_1 := 0.0
  part_2 := 0.0

  for _, mass := range masses {
	   fuel := calculateFuel(mass)

     part_1 += fuel
     part_2 += fuel

     last := fuel

     for {
       fuel_of_fuel := calculateFuel(last)

       if fuel_of_fuel <= 0 {
         break
       }

       part_2 += fuel_of_fuel
       last = fuel_of_fuel
     }
  }

  fmt.Printf("Part 1: %d\n", int(part_1))
  fmt.Printf("Part 2: %d\n", int(part_2))
}

func calculateFuel(fuel float64) float64 {
  return math.Floor(fuel / 3) - 2
}

func getInput() []float64 {
  file, err := os.Open("../input")
  if err != nil {
    log.Fatal(err)
  }

  defer file.Close()

  scanner := bufio.NewScanner(file)
  var masses []float64

  for scanner.Scan() {
    f, err := strconv.ParseFloat(scanner.Text(), 64)

    if err != nil {
      log.Fatal(err)
    }

    masses = append(masses, f)
  }

  return masses
}

File: day-1/python/main.py
"""Solution for AoC 2019 day 1."""

part1_fuel = 0
part2_fuel = 0


def calculate_fuel(mass: int) -> int:
    """Calculate required fuel given mass."""
    mass //= 3
    mass -= 2
    return mass


with open("../input") as f:
    for mass in f:
        fuel = calculate_fuel(int(mass))

        part1_fuel += fuel
        part2_fuel += fuel

        last = fuel
        while True:
            fuel_for_fuel = calculate_fuel(last)
            if fuel_for_fuel <= 0:
                break

            part2_fuel += fuel_for_fuel
            last = fuel_for_fuel

print(f"Part 1: {part1_fuel}")
print(f"Part 2: {part2_fuel}")
